### PR TITLE
Add In app Mint Flow

### PR DESCRIPTION
**Pull request by kaitoo1**
### Summary of Changes

Add App Claim Flow

### Demo or Before/After Pics

Shared demo on slack

### Edge Cases

List any common edge cases that you have considered and tested.

### Testing Steps
Point build to dev and open notifications, then tap the Mint Announcement notification

### Checklist

Please make sure to review and check all of the following:

- [x] I've tested the changes and all tests pass.
- [ ] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [x] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

File: apps/mobile/src/components/Mint/MintCampaign/MintCampaignPostTransaction.tsx [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+  }, [navigateToCommunity, onClose, token?.definition?.community]);
+
+  if (state === 'TOKEN_SYNCED' && token) {
+    return (
+      <View>
+        <View className="mb-1">
+          <TitleS>Congratulations!</TitleS>
+          <TitleS>You collected {`${token?.definition.name} by MCHX`}</TitleS>
+        </View>
+        {/* <BaseM>You have collected {`${token?.name} by MCHX`}</BaseM> */}
```
Comment (Robinnnnn) [inline apps/mobile/src/components/Mint/MintCampaign/MintCampaignPostTransaction.tsx:143]: can toss this?

File: apps/mobile/src/components/Mint/MintCampaign/MintCampaignPostTransaction.tsx [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    }
+    onClose();
+  }, [navigateToCommunity, onClose, token?.definition?.community]);
+
+  if (state === 'TOKEN_SYNCED' && token) {
+    return (
+      <View>
+        <View className="mb-1">
+          <TitleS>Congratulations!</TitleS>
+          <TitleS>You collected {`${token?.definition.name} by MCHX`}</TitleS>
```
Comment (Robinnnnn) [inline apps/mobile/src/components/Mint/MintCampaign/MintCampaignPostTransaction.tsx:141]: can use fallback: `token?.definition.name ?? 'Radiance'`

File: apps/mobile/src/components/Mint/MintCampaign/MintCampaignPostTransaction.tsx [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+    return () => clearInterval(interval);
+  }, [fadeIn, fadeOut]);
+  return (
+    <View className="text-center h-24">
+      <Animated.View style={[animatedStyle]}>
+        <BaseM classNameOverride="text-shadow text-center ">{COPY[index]}</BaseM>
+      </Animated.View>
+    </View>
+  );
```
Comment (Robinnnnn) [inline apps/mobile/src/components/Mint/MintCampaign/MintCampaignPostTransaction.tsx:248]: niiiiice thanks for animating 🔥🔥🔥

File: apps/mobile/src/components/Mint/MintCampaign/MintCampaignPreTransaction.tsx [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    return now > endDate;
+  }, []);
+
+  const [timeLeft, setTimeLeft] = useState(calculateTimeLeftText());
+  const [error, setError] = useState('');
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setTimeLeft(calculateTimeLeftText());
+    }, 60000);
```
Comment (Robinnnnn) [inline apps/mobile/src/components/Mint/MintCampaign/MintCampaignPreTransaction.tsx:51]: so after a minute passes, we run another calculation?

File: apps/mobile/src/components/Mint/MintCampaign/MintCampaignPostTransaction.tsx (hunk 2 above)
Comment (kaitoo1) [reply apps/mobile/src/components/Mint/MintCampaign/MintCampaignPostTransaction.tsx:141]: smartttt

File: apps/mobile/src/components/Mint/MintCampaign/MintCampaignPreTransaction.tsx [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+import { Button } from '~/components/Button';
+import { BaseM, BaseS, TitleLItalic, TitleS } from '~/components/Text';
+import { MintCampaignPreTransactionQuery } from '~/generated/MintCampaignPreTransactionQuery.graphql';
+
+import { MCHX_CLAIM_CODE_KEY } from './MintCampaignBottomSheet';
+
+export const MCHX_MINT_CAMPAIGN_END_DATE = '2024-05-05T10:00:00-04:00';
+
+const MCHX_COLLECTION_ID = '65ec1cefb5813a0adf4dff4e';
```
Comment (jarrel-b) [inline apps/mobile/src/components/Mint/MintCampaign/MintCampaignPreTransaction.tsx:16]: Change this to `660d4342c6bc04d5dc5598e7`

Comment (kaitoo1) [reply apps/mobile/src/components/Mint/MintCampaign/MintCampaignPreTransaction.tsx]: done

Comment (kaitoo1) [reply apps/mobile/src/components/Mint/MintCampaign/MintCampaignPreTransaction.tsx]: nvm you added! lol thx

**Pull request merged**